src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -14339,6 +14321,86 @@ static void verify(CSharpCompilation comp)
             }
         }
 
+        [Theory]
+        [InlineData(LanguageVersion.CSharp10)]
+        [InlineData(LanguageVersion.CSharp11)]
+        public void LocalScope_01_Foreach_Deconstruction_02(LanguageVersion languageVersion)
+        {
+            var source =
+@"#pragma warning disable 219
+
+class Program
+{
+    static void F(ref R r)
+    {
+        foreach ((R r0, _) in new Enumerable1()) break;
```

Comment:
Consider adding a non-deconstruction variant: `foreach (var _ in ...)`

Reply:
Also, I liked how for deconstruction tests you added the equivalent lowered version (calling `Deconstruct`). Consider doing the same for foreach-deconstruction scenarios, adding equivalent lowered code.